### DISCUSSION TITLE
middleware property in Nuxt 3?

### DISCUSSION THREAD

**Discussion by hitochan777** [nuxt/nuxt#25684] [Questions]
I am trying to migrate from Nuxt 2 to Nuxt 3.
My project uses [`middleware` property](https://v2.nuxt.com/docs/components-glossary/middleware/) in Nuxt 2, but it seems the property is not supported in Nuxt 3 even when I use Option API.
Is there any workaround?

**Comment by AloisSeckar** [comment]
In Nuxt 3 you associate middleware with a specific page(s) via [`definePageMeta` macro](https://nuxt.com/docs/api/utils/define-page-meta) in their setup:

```
// /pages/page.vue

<script setup lang="ts">
definePageMeta({
  middleware: 'named-middleware' // or an array of strings, if you need more
})
</script>
```

while you have the middleware handler:

```
// /middleware/named-middleware.ts

export default defineNuxtRouteMiddleware((to, from) => {
  // middleware logic here
}
```

**Reply by hitochan777** [reply]
Thanks.

I would like to apply certain middlewares to pages that use a specific layout.
I tried using the macro in a layout but got the following warning `definePageMeta() is a compiler-hint helper that is only usable inside the script block of a single file component which is also a page. Its arguments should be compiled away and passing it at runtime has no effect.`.
In Nuxt 2, I can specify middlewares in layouts using middleware props.
